internal/access/signup.go

```diff
@@ -13,38 +15,39 @@ import (
 
 // Signup creates a user identity using the supplied name and password and
 // grants the identity "admin" access to Infra.
-func Signup(c *gin.Context, name, password, orgName string) (*models.Identity, *models.Organization, error) {
+func Signup(c *gin.Context, keyExpiresAt time.Time, name, password string, org *models.Organization) (*models.Identity, string, error) {
 	// no authorization is setup yet
 	db := getDB(c)
 
-	org := &models.Organization{Name: orgName}
+	err := checkPasswordRequirements(db, password)
```

Comment:
Don't we need to do this after `CreateOrganization` , otherwise it's checking the password requirements for the default org, instead of the new org? Or is that by default? Maybe add a comment about that if it's by design?

I think this illustrates one of the disadvantages to just accepting org out of context. It makes behaviour like this less obvious. Although maybe we fix that later by not setting an org in the context for requests like this?

Reply:
I was bit torn on this part. I moved the check to after the org creation rather than some global minimum password requirements, that makes more sense.